src/platform/GenericOTARequestorDriver.h

```diff
@@ -42,23 +43,45 @@ class GenericOTARequestorDriver : public OTARequestorDriver
         mImageProcessor = processor;
     }
 
+    // Set the timeout (in seconds) for the Default Provider timer
+    void SetDefaultProvidersTimeoutSec(uint32_t timeout) { mDefaultProviderTimeoutSec = timeout; }
+
+    //// Virtual methods from OTARequestorDriver
     bool CanConsent() override;
     uint16_t GetMaxDownloadBlockSize() override;
-
     void HandleError(UpdateFailureState state, CHIP_ERROR error) override;
+    void HandleIdleState() override;
     void UpdateAvailable(const UpdateDescription & update, System::Clock::Seconds32 delay) override;
     void UpdateNotFound(UpdateNotFoundReason reason, System::Clock::Seconds32 delay) override;
     void UpdateDownloaded() override;
     void UpdateConfirmed(System::Clock::Seconds32 delay) override;
     void UpdateSuspended(System::Clock::Seconds32 delay) override;
     void UpdateDiscontinued() override;
     void UpdateCancelled() override;
+    void ScheduleDelayedAction(System::Clock::Seconds32 delay, System::TimerCompleteCallback action, void * aAppState) override;
+    void CancelDelayedAction(System::TimerCompleteCallback action, void * aAppState) override;
+    void ProcessAnnounceOTAProviders(const ProviderLocationType & providerLocation,
+                                     app::Clusters::OtaSoftwareUpdateRequestor::OTAAnnouncementReason announcementReason) override;
+    void DriverSendQuery() override;
+    // Determines the next available Provider location and sets it in the OTARequestor
+    void DetermineAndSetProviderLocation() override;
 
-protected:
-    void ScheduleDelayedAction(UpdateFailureState state, System::Clock::Seconds32 delay, System::TimerCompleteCallback action);
+    //// Regular methods
+    void StartDefaultProviderTimer();
+    void StopDefaultProviderTimer();
+    void DefaultProviderTimerHandler(System::Layer * systemLayer, void * appState);
+
+private:
+    // Returns the next available Provider location
+    bool DetermineProviderLocation(app::Clusters::OtaSoftwareUpdateRequestor::Structs::ProviderLocation::Type & providerLocation);
 
     OTARequestorInterface * mRequestor           = nullptr;
     OTAImageProcessorInterface * mImageProcessor = nullptr;
+    uint32_t mOtaStartDelayMs                    = 0;
+    uint32_t mDefaultProviderTimeoutSec          = 86400; // Timeout for the Default Provider timer
```

Comment:
nit: This value really represents the periodic query. Calling it `mDefaultProviderTimeoutSec` isn't very obvious what it really is meant to do. Likewise, `// Timeout for the Default Provider timer` doesn't really tell me what it represents without looking at the code above to find out. Suggest something like `mPeriodicQueryTimeInterval` and change the functions accordingly that handles this timer value.

Reply:
+1 on naming. And please express it as (24 * 60 * 60)

Reply:
Done